packages/opentelemetry-scope-base/src/context.ts

```diff
@@ -0,0 +1,72 @@
+/*!
+ * Copyright 2020, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/** Map of identifiers to an unknown value used internally to store context */
+type Store = { [identifer: string]: unknown };
+
+/**
+ * Class which stores and manages current context values. All methods which
+ * update context such as get and delete do not modify an existing context,
+ * but create a new one with updated values.
+ */
+export class Context {
+  private _context: Store;
+
+  public static readonly ROOT_CONTEXT = new Context();
```

Comment:
is it needed that this is public? If yes I would recommend to somehow document when `ROOT_CONTEXT` should be used and when the current context.

Reply:
The root context is used by the scope manager https://github.com/dynatrace-oss-contrib/opentelemetry-js/blob/38cda1d05214f1b2976c0b10ee1b1306077e70f3/packages/opentelemetry-scope-async-hooks/src/AsyncHooksScopeManager.ts#L55

Reply:
I will document this